sdk/vppagent/flush.go

```diff
@@ -0,0 +1,172 @@
+	"github.com/networkservicemesh/networkservicemesh/sdk/common"
+	"github.com/networkservicemesh/networkservicemesh/sdk/endpoint"
+	"github.com/opentracing/opentracing-go"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+// Flush is a VPP Agent Flush composite
+type Flush struct {
+	endpoint.BaseCompositeEndpoint
+	Conn     *grpc.ClientConn
+}
+
+// Request implements the request handler
+func (f *Flush) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*connection.Connection, error) {
+
+	if f.GetNext() == nil {
+		logrus.Fatal("The VPP Agent Flush composite requires that there is Next set")
+	}
+
+	incomingConnection, err := f.GetNext().Request(ctx, request)
+	if err != nil {
+		logrus.Errorf("Next request failed: %v", err)
+		return nil, err
+	}
+
+	opaque := f.GetNext().GetOpaque(incomingConnection)
+	if opaque == nil {
+		err = fmt.Errorf("received empty data from Next")
+		logrus.Errorf("Unable to find the DataChange: %v", err)
+		return nil, err
+	}
+	dataChange := opaque.(*ConnectionData).DataChange
+
+	logrus.Infof("Sending DataChange to VPP Agent: %v", dataChange)
+	err = f.send(ctx, dataChange)
+	if err != nil {
+		logrus.Errorf("Failed to send DataChange to VPP Agent: %v", err)
+		return nil, err
+	}
+
+	return incomingConnection, nil
+}
+
+// Close implements the close handler
+func (f *Flush) Close(ctx context.Context, connection *connection.Connection) (*empty.Empty, error) {
+	opaque := f.GetNext().GetOpaque(connection)
+	if opaque != nil {
+		dataChange := opaque.(*ConnectionData).DataChange
+
+		logrus.Infof("Removing DataChange from VPP Agent: %v", dataChange)
+		err := f.remove(ctx, dataChange)
+		if err != nil {
+			logrus.Errorf("Failed to remove DataChange from VPP Agent: %v", err)
+			return &empty.Empty{}, err
+		}
+	}
+
+	err := f.Conn.Close()
+	if err != nil {
+		logrus.Errorf("Failed to close connection to VPP Agent: %v", err)
+		return &empty.Empty{}, err
+	}
+
+	if f.GetNext() != nil {
+		return f.GetNext().Close(ctx, connection)
+	}
+	return &empty.Empty{}, nil
+}
+
+// Name returns the composite name
+func (f *Flush) Name() string {
+	return "flush"
+}
+
+// NewFlush creates a Flush
+func NewFlush(configuration *common.NSConfiguration, endpoint string) *Flush {
+	// ensure the env variables are processed
+	if configuration == nil {
+		configuration = &common.NSConfiguration{}
+	}
+	configuration.CompleteNSConfiguration()
+
+	conn, err := createConnection(endpoint)
+	if err != nil {
+		logrus.Errorf("Failed to create connection to VPP Agent: %v", err)
+		return nil
+	}
+
+	self := &Flush{
+		Conn: conn,
+	}
+
+	logrus.Infof("Resetting VPP Agent")
+	err = self.reset()
+	if err != nil {
+		logrus.Errorf("Failed to reset VPP Agent: %v", err)
+		return nil
+	}
+
+	return self
+}
+
+func createConnection(endpoint string) (*grpc.ClientConn, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 120*time.Second)
```

Comment:
Should we have timeout in constants or configurable somehow in pipeline?

Reply:
Added constants like it is done in other code in the repository